emu/src/cpu/arm/operations.rs

```diff
@@ -463,6 +463,18 @@ impl Arm7tdmi {
         self.cpsr.set_zero_flag(value == 0);
     }
 
+    /// Subtract the contents of rs from zero, and store the result in rd.
+    pub fn neg(&mut self, rd: usize, rs: u32) {
+        let u_result = 0u32.overflowing_sub(rs);
+        let i_result = 0i32.overflowing_sub(rs as i32);
+
+        self.registers.set_register_at(rd, u_result.0);
+        self.cpsr.set_carry_flag(!u_result.1);
+        self.cpsr.set_overflow_flag(i_result.1);
+        self.cpsr.set_zero_flag(u_result.0 == 0);
+        self.cpsr.set_sign_flag(i_result.0 > 0);
+    }
```

Comment:
I think here we can reuse `rsb(rd, rs, 0)`

Reply:
RSB does a SUB... I don't think is the semantically

Reply:
`neg` is the same as an `rsbs` in arm 

<img width="525" alt="image" src="https://github.com/RIP-Comm/clementine/assets/5256712/057079f0-a575-41ba-9bcc-f803197d2356">

Reply:
Fixed :D, at some point we should extract ALU